Exiled.API/Features/Toys/Light.cs

```diff
@@ -0,0 +1,102 @@
+// -----------------------------------------------------------------------
+// <copyright file="Light.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Features.Toys
+{
+    using AdminToys;
+    using Mirror;
+    using UnityEngine;
+
+    /// <summary>
+    /// A wrapper class for light source Admin Toys.
+    /// </summary>
+    public class Light
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Light"/> class.
+        /// </summary>
+        public Light()
+        {
+            Base = UnityEngine.Object.Instantiate(ToysHelper.LightBaseObject);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Light"/> class from a <see cref="LightSourceToy"/>.
+        /// </summary>
+        /// <param name="toy">The toy to be wrapped.</param>
+        public Light(LightSourceToy toy)
+        {
+            Base = toy;
+        }
+
+        /// <summary>
+        /// Gets the base <see cref="LightSourceToy"/>.
+        /// </summary>
+        public LightSourceToy Base { get; } = null;
+
+        /// <summary>
+        /// Gets or sets the position of the light.
+        /// </summary>
+        public Vector3 Position { get; set; } = Vector3.zero;
+
+        /// <summary>
+        /// Gets or sets the intensity of the light.
+        /// </summary>
+        public float Intensity
+        {
+            get => Base.NetworkLightIntensity;
+            set => Base.NetworkLightIntensity = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the range of the light.
+        /// </summary>
+        public float Range
+        {
+            get => Base.NetworkLightRange;
+            set => Base.NetworkLightRange = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the color of the primitive.
+        /// </summary>
+        public Color Color
+        {
+            get => Base.NetworkLightColor;
+            set => Base.NetworkLightColor = value;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the light should have shadows.
```

Comment:
Technically lights itself don't have shadows, I know what you mean in this but you can try changing it to something related to objects that will be illuminated by this light which is what this property do.

Reply:
I'm not entirely sure what this property does since I haven't tested it out. Any ideas for a good name?

Reply:
Gets or sets a value indicating whether the light should cause shadows from other objects.

ShadowEmission